kolibri/core/public/api.py

```diff
@@ -0,0 +1,88 @@
+import json
+import platform
+
+from django.db.models import Q
+from django.http import HttpResponse
+from django.http import HttpResponseNotFound
+from morango.models import InstanceIDModel
+from rest_framework import viewsets
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+import kolibri
+from .. import error_constants
+from kolibri.core.content.models import ChannelMetadata
+from kolibri.core.content.models import ContentNode
+from kolibri.core.content.serializers import PublicChannelSerializer
+
+
+class InfoViewSet(viewsets.ViewSet):
+    """
+    An equivalent endpoint in studio which allows kolibri devices to know
+    if this device can serve content.
+    Spec doc: https://docs.google.com/document/d/1XKXQe25sf9Tht6uIXvqb3T40KeY3BLkkexcV08wvR9M/edit#
+    """
+
+    def list(self, request):
+        """Returns metadata information about the device"""
+
+        instance_model = InstanceIDModel.get_or_create_current_instance()[0]
+
+        info = {'application': 'kolibri',
+                'kolibri_version': kolibri.__version__,
+                'instance_id': instance_model.id,
+                'device_name': instance_model.hostname,
+                'operating_system': platform.system()
+                }
+        return Response(info)
+
+
+def _get_channel_list(version, params, identifier=None):
+    if version == "v1":
+        return _get_channel_list_v1(params, identifier=identifier)
+    else:
+        raise LookupError()
```

Comment:
Don't know that we need to add the complexity of allowing for specifying other versions right now, since we don't know if we'll hit the point of needing to update that (and can add the logic in at that point). But if it's easier to leave, that's cool too.